### PR TITLE
SAK-33398 need to provide tools that implement ExternalAssignmentProvider context on the ExternalAssessment

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

**Comment by jonespm** [comment]
This sound like a great idea, thanks. This has a conflict that needs to be resolved.

**Comment by ottenhoff** [comment]
@ern can you look briefly at my assignment changes here?

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 1]
```diff
@@ -170,6 +171,19 @@ public String getLabel() {
     }
 
     @Override
+    public String getToolTitle() {
+        Tool tool = toolManager.getTool(AssignmentServiceConstants.ASSIGNMENT_TOOL_ID);
+        String toolTitle = null;
+
+        if (tool == null)
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: braces are preferrred

Comment (ottenhoff) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: agreed but it's just code copy

File: samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/AssessmentGradeInfoProvider.java [hunk 2]
```diff
@@ -110,15 +112,24 @@ private PublishedAssessmentIfc getPublishedAssessment(String id) {
         return a;
     }
 
-    public boolean isAssignmentDefined(String id) {
-        // SAM-3068 avoid looking up another tool's id
-        if (!StringUtils.isNumeric(id)) {
+    public boolean isAssignmentDefined(String externalAppName, String id) {
+    	    // SAM-3068 avoid looking up another tool's id
+    	    if (!StringUtils.isNumeric(id)) {
```
Comment (ern) [inline samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/AssessmentGradeInfoProvider.java]: this is deprecated use isCreatable()

Comment (ottenhoff) [reply samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/AssessmentGradeInfoProvider.java]: I don't see any deprecation in StringUtils:

https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNumeric-java.lang.CharSequence-

Comment (ern) [reply samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/integration/helper/integrated/AssessmentGradeInfoProvider.java]: yeah my bad was thinking NumberUtils.isNumeric

**Pull request merged**